components/dashboard/dashboard.go

```diff
@@ -217,7 +218,10 @@ func (d *Dashboard) Cleanup(cli client.Client, dscispec *dsciv1.DSCInitializatio
 	}
 
 	if shouldConfigureServiceMesh {
-		serviceMeshInitializer := feature.NewFeaturesInitializer(dscispec, d.defineServiceMeshFeatures(dscispec))
+		serviceMeshInitializer := feature.NewFeaturesInitializer(dscispec, d.defineServiceMeshFeatures(dscispec), featurev1.Origin{
```

Comment:
I would extract it to it's own `origin` variable for better readability. Also, if the only reason to pass it is to use it in `.For` of the `defineXYFeatures` I think it could be a parameter of this function instead. That would also remove a need for keeping it as a field of `FeaturesInitializer`.

Reply:
done here: 
0571826

learned about go's `closures` for this : ) , didn't know how to accomplish this at first which is what led to adding `origin` to the featureinitializer